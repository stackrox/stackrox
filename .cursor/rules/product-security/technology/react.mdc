---
description: Secure coding rules for React projects ⚛️
globs:
  - "**/*.jsx"
  - "**/*.tsx"
alwaysApply: false
---

- Use default React data binding to prevent XSS.
- Sanitize HTML before using dangerouslySetInnerHTML.
- Validate URLs before using them in the DOM.
- Secure server-side rendering and avoid JSON injection.
