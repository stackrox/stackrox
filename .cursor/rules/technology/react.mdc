# React-specific secure coding rules for Cursor
# This file provides security best practices for React projects.
---
description: Secure coding rules for React projects ⚛️
globs:
  - "**/*.jsx"
  - "**/*.tsx"
  - "**/src/**/*.jsx"
  - "**/src/**/*.tsx"
  - "**/components/**/*.jsx"
  - "**/components/**/*.tsx"
  - "**/pages/**/*.jsx"
  - "**/pages/**/*.tsx"
  - "**/App.jsx"
  - "**/App.tsx"
  - "**/index.jsx"
  - "**/index.tsx"
alwaysApply: false
---

- Use default React data binding to prevent XSS.
- Sanitize HTML before using dangerouslySetInnerHTML.
- Validate URLs before using them in the DOM.
- Secure server-side rendering and avoid JSON injection.
