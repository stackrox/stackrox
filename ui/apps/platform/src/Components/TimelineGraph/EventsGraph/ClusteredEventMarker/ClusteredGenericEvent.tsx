import React, { forwardRef } from 'react';

import { getNumEventsBackgroundWidth, getNumEventsText } from './clusteredEventMarkerUtils';

type ClusteredGenericEventProps = {
    size: number;
    numEvents: number;
};

const ClusteredGenericEvent = forwardRef<SVGSVGElement, ClusteredGenericEventProps>(
    ({ size, numEvents }, ref) => {
        const numEventsBackgroundWidth = getNumEventsBackgroundWidth(numEvents);
        const numEventsText = getNumEventsText(numEvents);
        return (
            <svg
                className="cursor-pointer"
                data-testid="clustered-generic-event"
                width={size}
                height={size}
                viewBox="0 0 23 22"
                xmlns="http://www.w3.org/2000/svg"
                ref={ref}
            >
                <g fill="none" fillRule="evenodd">
                    <rect fill="#4FAFD3" y="7" width="14.5" height="14.5" rx="7.25" />
                    <g transform="translate(9)">
                        <rect
                            stroke="#8FCFE7"
                            fill="#E2F7FF"
                            x=".5"
                            y=".5"
                            width={numEventsBackgroundWidth}
                            height="12"
                            rx="2"
                        />
                        <text
                            fontFamily="var(--pf-global--FontFamily--sans-serif)"
                            fontSize="9"
                            fontWeight="bold"
                            letterSpacing="-.18"
                            fill="#2A7895"
                        >
                            <tspan x="2.521" y="10">
                                {numEventsText}
                            </tspan>
                        </text>
                    </g>
                </g>
            </svg>
        );
    }
);

ClusteredGenericEvent.displayName = 'ClusteredGenericEvent';

export default ClusteredGenericEvent;
