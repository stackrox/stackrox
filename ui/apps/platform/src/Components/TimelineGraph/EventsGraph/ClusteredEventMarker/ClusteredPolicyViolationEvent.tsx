import React, { forwardRef } from 'react';

import { getNumEventsBackgroundWidth, getNumEventsText } from './clusteredEventMarkerUtils';

type ClusteredPolicyViolationEventProps = {
    size: number;
    numEvents: number;
};

const ClusteredPolicyViolationEvent = forwardRef<SVGSVGElement, ClusteredPolicyViolationEventProps>(
    ({ size, numEvents }, ref) => {
        const numEventsBackgroundWidth = getNumEventsBackgroundWidth(numEvents);
        const numEventsText = getNumEventsText(numEvents);
        return (
            <svg
                className="cursor-pointer"
                data-testid="clustered-policy-violation-event"
                width={size}
                height={size}
                viewBox="0 0 23 22"
                version="1.1"
                xmlns="http://www.w3.org/2000/svg"
                ref={ref}
            >
                <g fill="none" fillRule="evenodd">
                    <g transform="translate(0 6.1)">
                        <rect fill="#FF5782" width="14.5" height="14.5" rx="2.43" />
                        <path
                            d="M8.071 8.37H6.616l-.325-5.584h2.12L8.07 8.37zm-1.833 2.22c0-.346.091-.614.275-.804.184-.19.458-.286.82-.286.364 0 .639.094.825.282.186.189.279.458.279.808 0 .345-.096.613-.29.804-.192.19-.463.285-.813.285-.359 0-.631-.096-.817-.289-.186-.192-.28-.46-.28-.8z"
                            fill="#FFF"
                            fillRule="nonzero"
                        />
                    </g>
                    <g transform="translate(9)">
                        <rect
                            stroke="#D48D9F"
                            fill="#FFEBF0"
                            x=".5"
                            y=".5"
                            width={numEventsBackgroundWidth}
                            height="12"
                            rx="2"
                        />
                        <text
                            fontFamily="var(--pf-global--FontFamily--sans-serif)"
                            fontSize="9"
                            fontWeight="bold"
                            letterSpacing="-.18"
                            fill="#9D2E4B"
                        >
                            <tspan x="2.043" y="10">
                                {numEventsText}
                            </tspan>
                        </text>
                    </g>
                </g>
            </svg>
        );
    }
);

ClusteredPolicyViolationEvent.displayName = 'ClusteredPolicyViolationEvent';

export default ClusteredPolicyViolationEvent;
