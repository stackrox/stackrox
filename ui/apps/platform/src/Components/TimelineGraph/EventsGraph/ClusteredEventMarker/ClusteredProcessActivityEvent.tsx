import React, { forwardRef } from 'react';

import { getNumEventsBackgroundWidth, getNumEventsText } from './clusteredEventMarkerUtils';

type ClusteredProcessActivityEventProps = {
    inBaseline?: boolean;
    size: number;
    numEvents: number;
};

const ClusteredProcessActivityEvent = forwardRef<SVGSVGElement, ClusteredProcessActivityEventProps>(
    ({ inBaseline = false, size, numEvents }, ref) => {
        const numEventsBackgroundWidth = getNumEventsBackgroundWidth(numEvents);
        const numEventsText = getNumEventsText(numEvents);
        return inBaseline ? (
            <svg
                className="cursor-pointer"
                data-testid="clustered-process-in-baseline-activity-event"
                width={size}
                height={size}
                viewBox="0 0 23 22"
                xmlns="http://www.w3.org/2000/svg"
                ref={ref}
            >
                <g fill="none" fillRule="evenodd">
                    <g transform="translate(0 6)">
                        <rect fill="#56DDB2" width="14.58" height="14.58" rx="2.43" />
                        <path
                            d="M4.459 6.768a.807.807 0 00-1.18-.072.91.91 0 00-.067 1.24l2.685 3.17a.81.81 0 001.281-.043l4.645-6.244a.909.909 0 00-.144-1.233.808.808 0 00-1.172.151L6.48 9.153l-2.02-2.385z"
                            fill="#FFF"
                            fillRule="nonzero"
                        />
                    </g>
                    <g transform="translate(9)">
                        <rect
                            stroke="#8DD4BD"
                            fill="#EBFFF9"
                            x=".5"
                            y=".5"
                            width={numEventsBackgroundWidth}
                            height="12"
                            rx="2"
                        />
                        <text
                            fontFamily="var(--pf-global--FontFamily--sans-serif)"
                            fontSize="9"
                            fontWeight="bold"
                            letterSpacing="-.18"
                            fill="#2E9D7A"
                        >
                            <tspan x="2.043" y="10">
                                {numEventsText}
                            </tspan>
                        </text>
                    </g>
                </g>
            </svg>
        ) : (
            <svg
                className="cursor-pointer"
                data-testid="clustered-process-activity-event"
                width={size}
                height={size}
                viewBox="0 0 23 22"
                xmlns="http://www.w3.org/2000/svg"
                ref={ref}
            >
                <g fill="none" fillRule="evenodd">
                    <rect
                        width="14.5"
                        height="14.5"
                        rx="2.43"
                        transform="translate(0 6.1)"
                        fill="#5677DD"
                    />
                    <g transform="translate(9)">
                        <rect
                            stroke="#8D9FD4"
                            fill="#EBF0FF"
                            x=".5"
                            y=".5"
                            width={numEventsBackgroundWidth}
                            height="12"
                            rx="2"
                        />
                        <text
                            fontFamily="var(--pf-global--FontFamily--sans-serif)"
                            fontSize="9"
                            fontWeight="bold"
                            letterSpacing="-.18"
                            fill="#2E4A9D"
                        >
                            <tspan x="2.043" y="10">
                                {numEventsText}
                            </tspan>
                        </text>
                    </g>
                </g>
            </svg>
        );
    }
);

ClusteredProcessActivityEvent.displayName = 'ClusteredProcessActivityEvent';

export default ClusteredProcessActivityEvent;
