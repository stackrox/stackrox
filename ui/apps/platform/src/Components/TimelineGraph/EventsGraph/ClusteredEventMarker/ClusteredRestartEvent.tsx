import React, { forwardRef } from 'react';

import { getNumEventsBackgroundWidth, getNumEventsText } from './clusteredEventMarkerUtils';

type ClusteredRestartEventProps = {
    size: number;
    numEvents: number;
};

const ClusteredRestartEvent = forwardRef<SVGSVGElement, ClusteredRestartEventProps>(
    ({ size, numEvents }, ref) => {
        const numEventsBackgroundWidth = getNumEventsBackgroundWidth(numEvents);
        const numEventsText = getNumEventsText(numEvents);
        return (
            <svg
                className="cursor-pointer"
                data-testid="clustered-restart-event"
                width={size}
                height={size}
                viewBox="0 0 22 22"
                xmlns="http://www.w3.org/2000/svg"
                ref={ref}
                fillRule="evenodd"
                clipRule="evenodd"
            >
                <path
                    d="M9.531 7.708l6.848 11.67a1.402 1.402 0 01-.527 1.932c-.218.125-.466.19-.718.19H1.438A1.426 1.426 0 010 20.085c0-.248.066-.492.193-.707l6.848-11.67a1.449 1.449 0 012.49 0z"
                    fill="#ff9064"
                />
                <g transform="translate(8)">
                    <rect
                        stroke="#D87953"
                        fill="#FFEBE3"
                        x=".5"
                        y=".5"
                        width={numEventsBackgroundWidth}
                        height="12"
                        rx="2"
                    />
                    <text
                        fontFamily="var(--pf-global--FontFamily--sans-serif)"
                        fontSize="9"
                        fontWeight="bold"
                        letterSpacing="-.18"
                        fill="#8B4225"
                    >
                        <tspan x="2.043" y="10">
                            {numEventsText}
                        </tspan>
                    </text>
                </g>
            </svg>
        );
    }
);

ClusteredRestartEvent.displayName = 'ClusteredRestartEvent';

export default ClusteredRestartEvent;
