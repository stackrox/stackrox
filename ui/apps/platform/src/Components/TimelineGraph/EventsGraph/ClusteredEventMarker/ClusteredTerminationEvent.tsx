import React, { forwardRef } from 'react';

import { getNumEventsBackgroundWidth, getNumEventsText } from './clusteredEventMarkerUtils';

type ClusteredTerminationEventProps = {
    size: number;
    numEvents: number;
};

const ClusteredTerminationEvent = forwardRef<SVGSVGElement, ClusteredTerminationEventProps>(
    ({ size, numEvents }, ref) => {
        const numEventsBackgroundWidth = getNumEventsBackgroundWidth(numEvents);
        const numEventsText = getNumEventsText(numEvents);
        return (
            <svg
                className="cursor-pointer"
                data-testid="clustered-termination-event"
                width={size}
                height={size}
                viewBox="0 0 22 22"
                xmlns="http://www.w3.org/2000/svg"
                ref={ref}
            >
                <g fill="none" fillRule="evenodd">
                    <path
                        d="M9.53 20.792l6.849-11.67a1.402 1.402 0 00-.527-1.932 1.456 1.456 0 00-.718-.19H1.438C.644 7 0 7.633 0 8.415c0 .248.066.492.193.707l6.848 11.67a1.45 1.45 0 002.49 0z"
                        fill="#FF9064"
                    />
                    <g transform="translate(8)">
                        <rect
                            stroke="#D87953"
                            fill="#FFEBE3"
                            x=".5"
                            y=".5"
                            width={numEventsBackgroundWidth}
                            height="12"
                            rx="2"
                        />
                        <text
                            fontFamily="var(--pf-v5-global--FontFamily--sans-serif)"
                            fontSize="9"
                            fontWeight="bold"
                            letterSpacing="-.18"
                            fill="#8B4225"
                        >
                            <tspan x="2.043" y="10">
                                {numEventsText}
                            </tspan>
                        </text>
                    </g>
                </g>
            </svg>
        );
    }
);

ClusteredTerminationEvent.displayName = 'ClusteredTerminationEvent';

export default ClusteredTerminationEvent;
