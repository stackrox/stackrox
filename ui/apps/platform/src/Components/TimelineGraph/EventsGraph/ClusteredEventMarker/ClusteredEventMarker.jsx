import React, { useRef } from 'react';
import PropTypes from 'prop-types';
import { scaleLinear } from 'd3-scale';
import uniq from 'lodash/uniq';

import { getWidth } from 'utils/d3Utils';
import { clusteredEventTypes } from 'constants/timelineTypes';
import mainViewSelector from 'Components/TimelineGraph/MainView/selectors';
import D3Anchor from 'Components/D3Anchor';
import ClusteredEventsTooltip from 'Components/TimelineGraph/EventsGraph/ClusteredEventsTooltip';
import ClusteredGenericEvent from './ClusteredGenericEvent';
import ClusteredPolicyViolationEvent from './ClusteredPolicyViolationEvent';
import ClusteredRestartEvent from './ClusteredRestartEvent';
import ClusteredProcessActivityEvent from './ClusteredProcessActivityEvent';
import ClusteredTerminationEvent from './ClusteredTerminationEvent';

/**
 * Determines the type of the clustered event based on the group of events
 * @param {Object[]} events
 * @returns {clusteredEventTypes}
 */
function getClusterEventType(events) {
    const types = uniq(
        events.map((event) =>
            // if the event is a process in baseline activity, we should use a new type specific
            // to just clustered events
            event.inBaseline ? clusteredEventTypes.PROCESS_IN_BASELINE_ACTIVITY : event.type
        )
    );
    // if all the events are the same type, use that type
    if (types.length === 1) {
        return types[0];
    }
    // if we have multiple types of events, use the generic type
    return clusteredEventTypes.GENERIC;
}

const ClusteredEventMarker = ({
    events,
    differenceInMilliseconds,
    translateX,
    translateY,
    minTimeRange,
    maxTimeRange,
    size,
    margin,
}) => {
    const popoverRef = useRef(null);

    // the "container" argument is a reference to the container for the D3-related element
    function onUpdate(container) {
        const width = getWidth(mainViewSelector);
        const minRange = margin;
        const maxRange = width - margin;
        const xScale = scaleLinear()
            .domain([minTimeRange, maxTimeRange])
            .range([minRange, maxRange]);
        const x = xScale(differenceInMilliseconds).toFixed(0);

        container.attr(
            'transform',
            `translate(${Number(translateX) + Number(x) - size / 2}, ${
                Number(translateY) - size / 2
            })`
        );
    }

    const clusterEventType = getClusterEventType(events);
    const numEvents = events.length;

    return (
        <D3Anchor
            dataTestId="timeline-clustered-event-marker"
            translateX={translateX}
            translateY={translateY}
            onUpdate={onUpdate}
        >
            <ClusteredEventsTooltip events={events} popoverRef={popoverRef}>
                <g>
                    {clusterEventType === clusteredEventTypes.GENERIC && (
                        <ClusteredGenericEvent size={size} numEvents={numEvents} ref={popoverRef} />
                    )}
                    {clusterEventType === clusteredEventTypes.POLICY_VIOLATION && (
                        <ClusteredPolicyViolationEvent
                            size={size}
                            numEvents={numEvents}
                            ref={popoverRef}
                        />
                    )}
                    {clusterEventType === clusteredEventTypes.PROCESS_ACTIVITY && (
                        <ClusteredProcessActivityEvent
                            size={size}
                            numEvents={numEvents}
                            ref={popoverRef}
                        />
                    )}
                    {clusterEventType === clusteredEventTypes.PROCESS_IN_BASELINE_ACTIVITY && (
                        <ClusteredProcessActivityEvent
                            size={size}
                            inBaseline
                            numEvents={numEvents}
                            ref={popoverRef}
                        />
                    )}
                    {clusterEventType === clusteredEventTypes.RESTART && (
                        <ClusteredRestartEvent size={size} numEvents={numEvents} ref={popoverRef} />
                    )}
                    {clusterEventType === clusteredEventTypes.TERMINATION && (
                        <ClusteredTerminationEvent
                            size={size}
                            numEvents={numEvents}
                            ref={popoverRef}
                        />
                    )}
                </g>
            </ClusteredEventsTooltip>
        </D3Anchor>
    );
};

// TODO: replace with `Event` TS type once this module is migrated to TypeScript
const eventPropTypes = {
    name: PropTypes.string.isRequired,
    args: PropTypes.string,
    type: PropTypes.string.isRequired,
    uid: PropTypes.number,
    parentName: PropTypes.string,
    parentUid: PropTypes.number,
    reason: PropTypes.string,
    timestamp: PropTypes.string.isRequired,
    inBaseline: PropTypes.bool,
};

ClusteredEventMarker.propTypes = {
    events: PropTypes.arrayOf(PropTypes.shape(eventPropTypes)).isRequired,
    differenceInMilliseconds: PropTypes.number.isRequired,
    translateX: PropTypes.number.isRequired,
    translateY: PropTypes.number.isRequired,
    minTimeRange: PropTypes.number.isRequired,
    maxTimeRange: PropTypes.number.isRequired,
    size: PropTypes.number.isRequired,
    margin: PropTypes.number,
};

ClusteredEventMarker.defaultProps = {
    margin: 0,
};

export default ClusteredEventMarker;
