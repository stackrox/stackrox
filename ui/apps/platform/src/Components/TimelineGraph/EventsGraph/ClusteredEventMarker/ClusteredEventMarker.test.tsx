import React from 'react';
import { render, screen } from '@testing-library/react';

import { Event } from '../eventTypes';
import ClusteredEventMarker from './ClusteredEventMarker';

test('should show a clustered generic event marker', () => {
    const events: Event[] = Array.from(Array(9).keys()).map((index) => {
        return {
            id: `${index}`,
            name: `event-${index}`,
            type: 'PolicyViolationEvent',
            timestamp: '2020-04-20T20:20:20.358227916Z',
        };
    });
    events.push({
        id: '10',
        name: `event-10`,
        type: 'ProcessActivityEvent',
        args: '-g daemon off;',
        parentUid: -1,
        uid: 1000,
        timestamp: '2020-04-20T20:20:20.358227916Z',
    });
    const { asFragment } = render(
        <svg height={100} width={100} data-testid="timeline-main-view">
            <ClusteredEventMarker
                events={events}
                differenceInMilliseconds={50}
                translateX={0}
                translateY={25}
                minTimeRange={0}
                maxTimeRange={100}
                size={25}
            />
        </svg>
    );
    expect(screen.getByTestId('clustered-generic-event')).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
});

test('should show a clustered policy violation event marker', () => {
    const events = Array.from(Array(10).keys()).map((index) => {
        return {
            id: index,
            name: `event-${index}`,
            type: 'PolicyViolationEvent',
            timestamp: '2020-04-20T20:20:20.358227916Z',
        };
    });
    const { asFragment } = render(
        <svg height={100} width={100} data-testid="timeline-main-view">
            <ClusteredEventMarker
                events={events}
                differenceInMilliseconds={50}
                translateX={0}
                translateY={25}
                minTimeRange={0}
                maxTimeRange={100}
                size={25}
            />
        </svg>
    );
    expect(screen.getByTestId('clustered-policy-violation-event')).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
});

test('should show a clustered process activity event marker', () => {
    const events = Array.from(Array(10).keys()).map((index) => {
        return {
            id: index,
            name: `event-${index}`,
            type: 'ProcessActivityEvent',
            args: '-g daemon off;',
            parentName: null,
            parentUid: -1,
            uid: 1000,
            timestamp: '2020-04-20T20:20:20.358227916Z',
        };
    });
    const { asFragment } = render(
        <svg height={100} width={100} data-testid="timeline-main-view">
            <ClusteredEventMarker
                events={events}
                differenceInMilliseconds={50}
                translateX={0}
                translateY={25}
                minTimeRange={0}
                maxTimeRange={100}
                size={25}
            />
        </svg>
    );
    expect(screen.getByTestId('clustered-process-activity-event')).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
});

test('should show a clustered process in baseline activity event marker', () => {
    const events = Array.from(Array(10).keys()).map((index) => {
        return {
            id: index,
            name: `event-${index}`,
            type: 'ProcessActivityEvent',
            args: '-g daemon off;',
            parentName: null,
            parentUid: -1,
            uid: 1000,
            inBaseline: true,
            timestamp: '2020-04-20T20:20:20.358227916Z',
        };
    });
    const { asFragment } = render(
        <svg height={100} width={100} data-testid="timeline-main-view">
            <ClusteredEventMarker
                events={events}
                differenceInMilliseconds={50}
                translateX={0}
                translateY={25}
                minTimeRange={0}
                maxTimeRange={100}
                size={25}
            />
        </svg>
    );
    expect(screen.getByTestId('clustered-process-in-baseline-activity-event')).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
});

test('should show a clustered container restart event marker', () => {
    const events = Array.from(Array(10).keys()).map((index) => {
        return {
            id: index,
            name: `event-${index}`,
            type: 'ContainerRestartEvent',
            timestamp: '2020-04-20T20:20:20.358227916Z',
        };
    });
    const { asFragment } = render(
        <svg height={100} width={100} data-testid="timeline-main-view">
            <ClusteredEventMarker
                events={events}
                differenceInMilliseconds={50}
                translateX={0}
                translateY={25}
                minTimeRange={0}
                maxTimeRange={100}
                size={25}
            />
        </svg>
    );
    expect(screen.getByTestId('clustered-restart-event')).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
});

test('should show a container termination event marker', () => {
    const events = Array.from(Array(10).keys()).map((index) => {
        return {
            id: index,
            name: `event-${index}`,
            type: 'ContainerTerminationEvent',
            reason: 'Because of Covid-19',
            timestamp: '2020-04-20T20:20:20.358227916Z',
        };
    });
    const { asFragment } = render(
        <svg height={100} width={100} data-testid="timeline-main-view">
            <ClusteredEventMarker
                events={events}
                differenceInMilliseconds={50}
                translateX={0}
                translateY={25}
                minTimeRange={0}
                maxTimeRange={100}
                size={25}
            />
        </svg>
    );
    expect(screen.getByTestId('clustered-termination-event')).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
});
