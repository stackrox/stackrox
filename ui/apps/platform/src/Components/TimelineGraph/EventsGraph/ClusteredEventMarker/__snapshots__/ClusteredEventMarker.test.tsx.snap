// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should show a clustered container restart event marker 1`] = `
<DocumentFragment>
  <svg
    data-testid="timeline-main-view"
    height="100"
    width="100"
  >
    <g
      data-testid="timeline-clustered-event-marker"
      transform="translate(NaN, 12.5)"
    >
      <g>
        <svg
          class="cursor-pointer"
          clip-rule="evenodd"
          data-testid="clustered-restart-event"
          fill-rule="evenodd"
          height="25"
          viewBox="0 0 22 22"
          width="25"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M9.531 7.708l6.848 11.67a1.402 1.402 0 01-.527 1.932c-.218.125-.466.19-.718.19H1.438A1.426 1.426 0 010 20.085c0-.248.066-.492.193-.707l6.848-11.67a1.449 1.449 0 012.49 0z"
            fill="#ff9064"
          />
          <g
            transform="translate(8)"
          >
            <rect
              fill="#FFEBE3"
              height="12"
              rx="2"
              stroke="#D87953"
              width="13"
              x=".5"
              y=".5"
            />
            <text
              fill="#8B4225"
              font-family="var(--pf-global--FontFamily--sans-serif)"
              font-size="9"
              font-weight="bold"
              letter-spacing="-.18"
            >
              <tspan
                x="2.043"
                y="10"
              >
                9+
              </tspan>
            </text>
          </g>
        </svg>
      </g>
    </g>
  </svg>
</DocumentFragment>
`;

exports[`should show a clustered generic event marker 1`] = `
<DocumentFragment>
  <svg
    data-testid="timeline-main-view"
    height="100"
    width="100"
  >
    <g
      data-testid="timeline-clustered-event-marker"
      transform="translate(NaN, 12.5)"
    >
      <g>
        <svg
          class="cursor-pointer"
          data-testid="clustered-generic-event"
          height="25"
          viewBox="0 0 23 22"
          width="25"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            fill="none"
            fill-rule="evenodd"
          >
            <rect
              fill="#4FAFD3"
              height="14.5"
              rx="7.25"
              width="14.5"
              y="7"
            />
            <g
              transform="translate(9)"
            >
              <rect
                fill="#E2F7FF"
                height="12"
                rx="2"
                stroke="#8FCFE7"
                width="13"
                x=".5"
                y=".5"
              />
              <text
                fill="#2A7895"
                font-family="var(--pf-global--FontFamily--sans-serif)"
                font-size="9"
                font-weight="bold"
                letter-spacing="-.18"
              >
                <tspan
                  x="2.521"
                  y="10"
                >
                  9+
                </tspan>
              </text>
            </g>
          </g>
        </svg>
      </g>
    </g>
  </svg>
</DocumentFragment>
`;

exports[`should show a clustered policy violation event marker 1`] = `
<DocumentFragment>
  <svg
    data-testid="timeline-main-view"
    height="100"
    width="100"
  >
    <g
      data-testid="timeline-clustered-event-marker"
      transform="translate(NaN, 12.5)"
    >
      <g>
        <svg
          class="cursor-pointer"
          data-testid="clustered-policy-violation-event"
          height="25"
          version="1.1"
          viewBox="0 0 23 22"
          width="25"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            fill="none"
            fill-rule="evenodd"
          >
            <g
              transform="translate(0 6.1)"
            >
              <rect
                fill="#FF5782"
                height="14.5"
                rx="2.43"
                width="14.5"
              />
              <path
                d="M8.071 8.37H6.616l-.325-5.584h2.12L8.07 8.37zm-1.833 2.22c0-.346.091-.614.275-.804.184-.19.458-.286.82-.286.364 0 .639.094.825.282.186.189.279.458.279.808 0 .345-.096.613-.29.804-.192.19-.463.285-.813.285-.359 0-.631-.096-.817-.289-.186-.192-.28-.46-.28-.8z"
                fill="#FFF"
                fill-rule="nonzero"
              />
            </g>
            <g
              transform="translate(9)"
            >
              <rect
                fill="#FFEBF0"
                height="12"
                rx="2"
                stroke="#D48D9F"
                width="13"
                x=".5"
                y=".5"
              />
              <text
                fill="#9D2E4B"
                font-family="var(--pf-global--FontFamily--sans-serif)"
                font-size="9"
                font-weight="bold"
                letter-spacing="-.18"
              >
                <tspan
                  x="2.043"
                  y="10"
                >
                  9+
                </tspan>
              </text>
            </g>
          </g>
        </svg>
      </g>
    </g>
  </svg>
</DocumentFragment>
`;

exports[`should show a clustered process activity event marker 1`] = `
<DocumentFragment>
  <svg
    data-testid="timeline-main-view"
    height="100"
    width="100"
  >
    <g
      data-testid="timeline-clustered-event-marker"
      transform="translate(NaN, 12.5)"
    >
      <g>
        <svg
          class="cursor-pointer"
          data-testid="clustered-process-activity-event"
          height="25"
          viewBox="0 0 23 22"
          width="25"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            fill="none"
            fill-rule="evenodd"
          >
            <rect
              fill="#5677DD"
              height="14.5"
              rx="2.43"
              transform="translate(0 6.1)"
              width="14.5"
            />
            <g
              transform="translate(9)"
            >
              <rect
                fill="#EBF0FF"
                height="12"
                rx="2"
                stroke="#8D9FD4"
                width="13"
                x=".5"
                y=".5"
              />
              <text
                fill="#2E4A9D"
                font-family="var(--pf-global--FontFamily--sans-serif)"
                font-size="9"
                font-weight="bold"
                letter-spacing="-.18"
              >
                <tspan
                  x="2.043"
                  y="10"
                >
                  9+
                </tspan>
              </text>
            </g>
          </g>
        </svg>
      </g>
    </g>
  </svg>
</DocumentFragment>
`;

exports[`should show a clustered process in baseline activity event marker 1`] = `
<DocumentFragment>
  <svg
    data-testid="timeline-main-view"
    height="100"
    width="100"
  >
    <g
      data-testid="timeline-clustered-event-marker"
      transform="translate(NaN, 12.5)"
    >
      <g>
        <svg
          class="cursor-pointer"
          data-testid="clustered-process-in-baseline-activity-event"
          height="25"
          viewBox="0 0 23 22"
          width="25"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            fill="none"
            fill-rule="evenodd"
          >
            <g
              transform="translate(0 6)"
            >
              <rect
                fill="#56DDB2"
                height="14.58"
                rx="2.43"
                width="14.58"
              />
              <path
                d="M4.459 6.768a.807.807 0 00-1.18-.072.91.91 0 00-.067 1.24l2.685 3.17a.81.81 0 001.281-.043l4.645-6.244a.909.909 0 00-.144-1.233.808.808 0 00-1.172.151L6.48 9.153l-2.02-2.385z"
                fill="#FFF"
                fill-rule="nonzero"
              />
            </g>
            <g
              transform="translate(9)"
            >
              <rect
                fill="#EBFFF9"
                height="12"
                rx="2"
                stroke="#8DD4BD"
                width="13"
                x=".5"
                y=".5"
              />
              <text
                fill="#2E9D7A"
                font-family="var(--pf-global--FontFamily--sans-serif)"
                font-size="9"
                font-weight="bold"
                letter-spacing="-.18"
              >
                <tspan
                  x="2.043"
                  y="10"
                >
                  9+
                </tspan>
              </text>
            </g>
          </g>
        </svg>
      </g>
    </g>
  </svg>
</DocumentFragment>
`;

exports[`should show a container termination event marker 1`] = `
<DocumentFragment>
  <svg
    data-testid="timeline-main-view"
    height="100"
    width="100"
  >
    <g
      data-testid="timeline-clustered-event-marker"
      transform="translate(NaN, 12.5)"
    >
      <g>
        <svg
          class="cursor-pointer"
          data-testid="clustered-termination-event"
          height="25"
          viewBox="0 0 22 22"
          width="25"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            fill="none"
            fill-rule="evenodd"
          >
            <path
              d="M9.53 20.792l6.849-11.67a1.402 1.402 0 00-.527-1.932 1.456 1.456 0 00-.718-.19H1.438C.644 7 0 7.633 0 8.415c0 .248.066.492.193.707l6.848 11.67a1.45 1.45 0 002.49 0z"
              fill="#FF9064"
            />
            <g
              transform="translate(8)"
            >
              <rect
                fill="#FFEBE3"
                height="12"
                rx="2"
                stroke="#D87953"
                width="13"
                x=".5"
                y=".5"
              />
              <text
                fill="#8B4225"
                font-family="var(--pf-global--FontFamily--sans-serif)"
                font-size="9"
                font-weight="bold"
                letter-spacing="-.18"
              >
                <tspan
                  x="2.043"
                  y="10"
                >
                  9+
                </tspan>
              </text>
            </g>
          </g>
        </svg>
      </g>
    </g>
  </svg>
</DocumentFragment>
`;
